cmd/metal-api/internal/metal/size.go

```diff
@@ -95,10 +95,10 @@ nextsize:
 	}
 
 	if len(found) == 0 {
-		return nil, matchlog, NotFound("no size found for hardware (%s)", hardware.ReadableSpec())
```

Comment:
There is also some defaulting applied in the `registerMachine` func. Is this really an error then?

Reply:
No actually a "nil" size is returned

Reply:
I think this is because there is no "unknown" size in the database

Reply:
Mh... I just think that returning an error + some ID is a little confusing for the caller. This is the caller:

```
	machineHardware := v1.NewMetalMachineHardware(&requestPayload.Hardware)
	size, _, err := r.ds.FromHardware(machineHardware)
	if err != nil {
		size = metal.UnknownSize
		utils.Logger(request).Sugar().Errorw("no size found for hardware, defaulting to unknown size", "hardware", machineHardware, "error", err)
	}
``` 

To me it feels better if we continue returning `nil` and in the code fragment above from `registerMachine` checks if the error was `NotFound` and then use the default there. In case it's not the `NotFound` error, the registration should fail and return 422 to the metal-hammer (undefined behavior). Should also help to explain my comment beneath a little better.